src/Processors/Formats/Impl/IonRowOutputFormat.cpp

```diff
@@ -0,0 +1,323 @@
+#include <Processors/Formats/Impl/IonRowOutputFormat.h>
+
+#if USE_ION
+
+#include <Formats/FormatFactory.h>
+#include <Common/assert_cast.h>
+
+#include <IO/WriteBufferFromString.h>
+#include <IO/WriteHelpers.h>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeMap.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeTuple.h>
+
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnLowCardinality.h>
+#include <Columns/ColumnMap.h>
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnTuple.h>
+#include "Formats/FormatSettings.h"
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_COLUMN;
+extern const int BAD_ARGUMENTS;
+}
+
+IonRowOutputFormat::IonRowOutputFormat(WriteBuffer & out_, const Block & header_, const FormatSettings & format_settings_)
+    : IRowOutputFormat(header_, out_)
+    , format_settings(format_settings_)
+    , writer(std::make_unique<IonWriter>(out, format_settings.ion.output_type == FormatSettings::IonOutputWriterType::BINARY))
+    , names(header_.getNames())
+{
+}
+
+void IonRowOutputFormat::serializeField(const IColumn & column, DataTypePtr data_type, size_t row_num)
+{
+    switch (data_type->getTypeId())
+    {
+        case TypeIndex::UInt8: {
+            writer->writeInt(assert_cast<const ColumnUInt8 &>(column).getElement(row_num));
```

Comment:
We also have `Bool` data type that has `TypeIdex::UInt8` (as it's implemented as `UInt8`). And we can process it separately:
```
if (isBool(data_type))
    writer->writeBool(assert_cast<const ColumnUInt8 &>(column).getElement(row_num));
else
    writer->writeInt(assert_cast<const ColumnUInt8 &>(column).getElement(row_num));
```